utilities/user_utils.py

```
@@ -59,16 +58,15 @@ def create_htpasswd_file(username: str, password: str) -> tuple[Path, str]:
-        temp_path = Path(temp_file.name).resolve()  # Get absolute path
-        run_command(command=shlex.split(f"htpasswd -c -b {temp_path.absolute()!s} {username} {password}"), check=True)
+    hashed = bcrypt.hashpw(password.encode(), bcrypt.gensalt()).decode()
+    htpasswd_line = f"{username}:{hashed}\n"
 
-        # Read the htpasswd file content and encode it
-        temp_file.seek(0)
-        htpasswd_content = temp_file.read()
-        htpasswd_b64 = base64.b64encode(htpasswd_content.encode()).decode()
+    with tempfile.NamedTemporaryFile(mode="w", delete=False, suffix=".htpasswd") as f:
+        f.write(htpasswd_line)
+        temp_path = Path(f.name)  # noqa: FCN001
```

Ruff (0.15.10)

[warning] Invalid rule code in `# noqa`: FCN001

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)